app/src/marketplace/containers/ProductController/ValidationContextProvider.jsx

```diff
@@ -42,6 +48,20 @@ function useValidationContext(): ContextProps {
 
     const isMounted = useIsMounted()
 
+    const setPendingChange = useCallback((name: string, isPending: boolean = true): Object => {
+        if (!isMounted()) { return }
+        if (!name) {
+            throw new Error('pending change needs a name')
+        }
+
+        setPendingChanges((state) => ({
+            ...state,
+            [name]: isPending,
+        }))
+    }, [setPendingChanges, isMounted])
+    const isPendingChange = useCallback((name: string) => !!pendingChanges[name], [pendingChanges])
+    const isAnyChangePending = useCallback(() => Object.values(pendingChanges).some(Boolean), [pendingChanges])
```

Comment:
Not too familiar with the code so this was a little confusing - what's the difference between `isPendingChange` and `isAnyChangePending`?

Reply:
`isPendingChange` is meant for individial fields so you can ask eg. if product field has been updated since last publish (such as `isPendingChange('name')`) whereas `isAnyChangePending` tells if any field has been changed and requires re-publish.

Reply:
ahh, gotcha 👍